presto-orc/src/main/java/com/facebook/presto/orc/metadata/OrcMetadataReader.java

```diff
@@ -388,8 +392,44 @@ private static BinaryStatistics toBinaryStatistics(OrcProto.BinaryStatistics bin
         return new BinaryStatistics(binaryStatistics.getSum());
     }
 
+    static Slice toSlice(ByteString value)
+    {
+        return Slices.wrappedBuffer(value.toByteArray());
+    }
+
+    @VisibleForTesting
+    static Slice maxStringTruncateAtFirstReplacementCharacter(Slice value)
+    {
+        if (value == null) {
+            return null;
+        }
+
+        int index = value.indexOf(REPLACEMENT_CHARACTER_UTF8);
+        if (index == -1) {
+            return value;
+        }
+        // Append 0xFF so that it is larger than value
+        Slice newValue = Slices.copyOf(value, 0, index + 1);
+        newValue.setByte(index, 0xFF);
+        return newValue;
+    }
+
+    @VisibleForTesting
+    static Slice minStringTruncateAtFirstReplacementCharacter(Slice value)
+    {
+        if (value == null) {
+            return null;
+        }
+
+        int index = value.indexOf(REPLACEMENT_CHARACTER_UTF8);
+        if (index == -1) {
+            return value;
+        }
+        return Slices.copyOf(value, 0, index);
+    }
+
     @VisibleForTesting
-    public static Slice getMaxSlice(String maximum)
+    static Slice getMaxSlice(String maximum)
```

Comment:
rename to `maxStringTruncateAtFirstSurrogateCharacter`?

The existing name is hardly informative.